subsys/pm/device.c

```diff
@@ -44,6 +44,10 @@ int pm_device_action_run(const struct device *dev,
 	struct pm_device *pm = dev->pm;
 	int ret;
 
+	if (!device_is_ready(dev)) {
```

Comment:
We don't do this in Zephyr. Any device client needs to make sure the device is ready before using it.

Reply:
`pm_device_action_run` expects a ready device, as any public Zephyr API does unless specified.

Reply:
Removed.